packages/test/src/test-local-activities.ts

```diff
@@ -1,339 +1,619 @@
-import anyTest, { TestFn } from 'ava';
+import { randomUUID } from 'crypto';
 import { firstValueFrom, Subject } from 'rxjs';
-import { v4 as uuid4 } from 'uuid';
-import { ApplicationFailure, defaultPayloadConverter, WorkflowClient, WorkflowFailedError } from '@temporalio/client';
+import { ExecutionContext, TestFn } from 'ava';
+import { Context as ActivityContext } from '@temporalio/activity';
+import {
+  ApplicationFailure,
+  defaultPayloadConverter,
+  WorkflowFailedError,
+  WorkflowHandle,
+  WorkflowStartOptions,
+} from '@temporalio/client';
+import { LocalActivityOptions } from '@temporalio/common';
+import { msToNumber } from '@temporalio/common/lib/time';
 import { temporal } from '@temporalio/proto';
-import { bundleWorkflowCode, WorkflowBundleWithSourceMap as WorkflowBundle } from '@temporalio/worker';
-import { isCancellation } from '@temporalio/workflow';
-import * as activities from './activities';
-import { RUN_INTEGRATION_TESTS, Worker } from './helpers';
-import * as workflows from './workflows/local-activity-testers';
+import { TestWorkflowEnvironment, workflowInterceptorModules } from '@temporalio/testing';
+import {
+  appendDefaultInterceptors,
+  bundleWorkflowCode,
+  DefaultLogger,
+  LogLevel,
+  Runtime,
+  WorkflowBundle,
+  WorkerOptions,
+} from '@temporalio/worker';
+import * as workflow from '@temporalio/workflow';
+import { test as anyTest, bundlerOptions, Worker } from './helpers';
+import { ConnectionInjectorInterceptor } from './activities/interceptors';
+
+// FIXME MOVE THIS SECTION SOMEWHERE IT CAN BE SHARED //
 
 interface Context {
+  env: TestWorkflowEnvironment;
   workflowBundle: WorkflowBundle;
-  taskQueue: string;
-  client: WorkflowClient;
-  getWorker: () => Promise<Worker>;
 }
 
 const test = anyTest as TestFn<Context>;
 
-test.before(async (t) => {
-  t.context.workflowBundle = await bundleWorkflowCode({
-    workflowsPath: require.resolve('./workflows/local-activity-testers'),
-  });
-});
+interface Helpers {
+  taskQueue: string;
+  createWorker(opts?: Partial<WorkerOptions>): Promise<Worker>;
+  executeWorkflow<T extends () => Promise<any>>(workflowType: T): Promise<workflow.WorkflowResultType<T>>;
+  executeWorkflow<T extends workflow.Workflow>(
+    fn: T,
+    opts: Omit<WorkflowStartOptions<T>, 'taskQueue' | 'workflowId'>
+  ): Promise<workflow.WorkflowResultType<T>>;
+  startWorkflow<T extends () => Promise<any>>(workflowType: T): Promise<WorkflowHandle<T>>;
+  startWorkflow<T extends workflow.Workflow>(
+    fn: T,
+    opts: Omit<WorkflowStartOptions<T>, 'taskQueue' | 'workflowId'>
+  ): Promise<WorkflowHandle<T>>;
+}
 
-test.beforeEach(async (t) => {
-  const title = t.title.replace('beforeEach hook for ', '');
-  const taskQueue = `test-local-activities-${title}`;
-  t.context = {
-    ...t.context,
-    client: new WorkflowClient(),
+function helpers(t: ExecutionContext<Context>): Helpers {
```

Comment:
Let's make sure we share this code.

Reply:
Definitely my intention. But that will be a different PR, after 1.8.0 release.